tests/testthat/test-pagination.R

```diff
@@ -178,3 +178,94 @@ test_that("cell and column wrapping works in pagination", {
         2L # Repeated rowname for context and its wrapping
     expect_identical(paginated_lines, tot_lines_w)
 })
+
+test_that("Pagination works with section dividers", {
+    lyt <- basic_table(title = "big title")  %>%
+        split_rows_by("SEX", section_div = "~") %>%
+        split_rows_by("ARM") %>%
+        analyze("AGE")
+
+    tt <- build_table(lyt, DM)
+
+    ttlst <- paginate_table(tt, lpp = 20)
+
+    expect_identical(length(ttlst), 2L)
+
+    expect_identical(tt[1:14,
+                        keep_titles = TRUE,
+                        reindex_refs = FALSE
+    ], ttlst[[1]])
+
+    expect_identical(
+        export_as_txt(ttlst[[1]][7:8, ]),
+        "big title\n\n——————————————\n       all obs\n——————————————\nMean    34.89 \n~~~~~~~~~~~~~~\nM             \n"
+    )
+
+    expect_identical(
+        paste0(export_as_txt(tail(ttlst[[1]], 1)), export_as_txt(head(ttlst[[2]], 1))),
+        paste0(
+            "big title\n\n——————————————\n       all obs\n——————————————\nMean    34.28 \n",
+            "big title\n\n———————————\n    all obs\n———————————\nU          \n"
+        )
+    )
+})
+
+test_that("Pagination works with non-default min_siblings", {
+    lyt <- basic_table() %>%
+        analyze("RACE")
+
+    tt <- build_table(lyt, DM)
+
+    ttlst <- expect_silent(paginate_table(tt, lpp = 3, min_siblings = 0))
+    expect_identical(length(ttlst), nlevels(DM$RACE))
+    expect_identical(tt[1], ttlst[[1]])
+    
+    expect_error(
+        paginate_table(tt, lpp = 3, min_siblings = 1),
+        "Unable to find any valid pagination between 1 and 1"
+    )
+})
+
+test_that("Pagination works with wrapped titles/footers", {
+    lyt <- basic_table() %>%
+        split_cols_by("SEX") %>%
+        analyze("RACE")
+    
+    tt <- build_table(lyt, DM)
+    
+    main_title(tt) <- "title with a\nnewline"
+    main_footer(tt) <- "wrapped footer with\nnewline"
+    
+    res <- expect_silent(paginate_table(tt, cpp = 60, tf_wrap = TRUE))
+    expect_identical(main_title(res[[1]]), main_title(res[[2]]))
+    expect_identical(main_title(res[[1]]), main_title(tt))
+    expect_identical(main_footer(res[[1]]), main_footer(res[[2]]))
+    expect_identical(main_footer(res[[1]]), main_footer(tt))
+    
+    main_title(tt) <- "this is a long long table title that should be wrapped to a new line"
+    main_footer(tt) <- "this is an extra long table main footer and should also be wrapped"
+    
+    res2 <- expect_silent(paginate_table(tt, cpp = 60, tf_wrap = TRUE))
+    expect_equal(length(res2), 2)
+    nrow_res2 <- 16 + 2
```

Comment:
Where did this 16 come from? If that is the number of lines the table takes up without tf_wrap, please derive that programmatically rather than hardcoding.